docs/DeveloperGuide.md

```diff
@@ -1,37 +1,209 @@
+
+### Common Classes
+
+Classes that are used by multiple components are placed in the `Common` package.
+
+
+## Implementation
+
+### **AddPatientCommand**
+
+The add patient feature allows users to add a new patient to the hospital system. This is facilitated by the `AddPatientCommand`, which handles the logic of adding a patient and updating the system’s state and storage.
 
-## Design & implementation
+1. **User Input**: The user enters the `addPatient` command followed by patient details (e.g., name).
+2. **Command Parsing**: The `Parser` parses the input and creates an `AddPatientCommand` object.
+3. **Execution**: The `AddPatientCommand` checks if the patient already exists in the `Hospital`. If not, it adds the patient.
+4. **Storage Update**: The updated hospital data, now containing the new patient, is saved to storage.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Sequence Diagram
 
+The following sequence diagram illustrates how the `AddPatientCommand` is executed:
+
+![AddPatientCommand Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddPatientSequenceDiagram.png)
+
+- **If Condition**: The command checks if the patient already exists before adding the patient.
+- **Logging**: If a duplicate is detected, an error is logged.
+
+### **AddTaskCommand**
+
+The add Task feature allows users to add different types of task to a selected patient's records. This is facilitated by the `AddTaskCommand`, which handles the logic of adding a task and provides feedback on the success or failure of task addition. The feature is only available when the user has navigated to the `TASK_STATE` by selecting a patient.
+
+There are three possible types of tasks - Todo, Deadline and Repeat.
+
+The sequence to add tasks involves:
+1. **User Input**:
+* **Todo**: The user enters the `todo` command followed by tag details (e.g., /tag).
+* **Deadline**: The user enters the `deadline` command followed by the deadline (e.g., /by) and tag details (e.g., /tag).
+* * **Repeat**: The user enters the `repeat` command followed by the recurring basis (e.g., /every) and tag details (e.g., /tag).
+2. **Command Parsing**:
+* **Todo Task**: The `Parser` parses the input and creates an `AddTodoParser` object.
+* **Deadline Task**: The `Parser` parses the input and creates an `AddDeadlineParser` object.
+* **Repeat Task**: The `Parser` parses the input and creates an `AddRepeatParser` object.
+3. **Execution**: The `AddTaskCommand` adds the task.
+4. **Storage Update**: The updated patient's data, now containing the new task, is saved to storage.
+
+
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how `Parser` parse command for `AddTaskCommand`:
+
+![Parse_Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParseAddTaskClassDiagram.png)
+
```

Comment:
I like how sequence diagrams are separated by levels of details and abstraction. Really helps in making diagrams that are easy to understand!